presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -840,12 +844,15 @@ private void invalidatePartitionCache(String databaseName, String tableName)
                         }
                     }));
         }
-        catch (NoSuchObjectException e) {
+        catch (HiveMetastoreClientNoSuchObjectException e) {
             // assume none of the partitions in the batch are available
             return stream(partitionNames.spliterator(), false)
                     .collect(toMap(identity(), (name) -> Optional.empty()));
         }
-        catch (TException e) {
+        catch (HiveMetastoreClientException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (IOException e) {
```

Comment:
This is because the ```HiveMetastoreClient``` interface extends ```Closable``` and ```close``` throws ```IOException``` and since we are calling an instance of the ```HiveMetastoreClient``` it will throw. I can add ```close``` to the interface without the exception so these don't have to be caught.